docs/glossary.rst

```diff
@@ -0,0 +1,102 @@
+:desc: Glossary for all Rasa-related terms
+
+.. _glossary:
+
+Glossary
+========
+
+.. glossary::
+
+    (Un)Happy Paths
+        Happy paths refer to conversations where everything proceeds smoothly as intended, and there are no deviations. Unhappy paths involve things like the user changing the topic conversation, correcting a previous input, etc.
+
+    Action
+        A single step a bot takes in a conversation (e.g. call an API or send a response).
+
+    Annotation
+        Labeling data.
+
+    CMS
+        Content Management System (CMS) can be used to store responses externally instead of hardcoding it as part of the domain.
+
+    Conversation
+        Actual user-bot messages - a back and forth between a user and a bot (so usually just messages - unlabeled conversation).
+
+    Custom Action
+        An action written by a Rasa developer that can run arbitrary code mainly to interact with the outside world.
+
+    Default Actions
+        Built-in actions that carry a special meaning to Rasa.
+
+    Domain
+        A domain lists all the intents, entities, and actions of an assistant.
+
+    Entities
+        Structured information that can be extracted from a user message (e.g. telephone number).
+
+    Event
+        Conversations in Rasa are represented as a sequence of events. 
+
+    Forms
+        One of the most common conversation patterns is to collect a few pieces of information from a user in order to do something (book a restaurant, call an API, search a database, etc.). This is also called slot filling.
+
+    Intent
+        Meaning of a users message selected as one of a set of predefined meanings (e.g. greeting).
+
+    Interactive Learning
+        You provide feedback to your bot while you talk to it. This is a powerful way to explore what your bot can do, and the easiest way to fix any mistakes it make.
+
+    NLG
+        Natural Language Generation (NLG) is the component that generates natural language output from structured data.
+
+    NLU
+        Natural Language Understanding (NLU) deals with parsing and understanding human language into a structured format. Rasa NLU is the part of Rasa that performs intent classification and entity extraction.
+
+    Open Source Software
```

Comment:
there are major arguments about the correct definition of open source, and I don't think we should invite those discussions here. I would exclude this from our glossary